interfaces/apparmor/backend.go

```diff
@@ -63,6 +70,138 @@ func (b *Backend) Name() interfaces.SecuritySystem {
+	// transparent to apparmor we must alter our profile to counter that and
+	// allow snap-confine to work.
+	if nfs, err := isHomeUsingNFS(); err != nil {
+		logger.Noticef("cannot determine if NFS is in use, %v", err)
+	} else if nfs {
+		policy["generated-nfs"] = &osutil.FileState{
+			Content: []byte(nfsSnippet),
+			Mode:    0644,
+		}
+		logger.Noticef("snapd enabled NFS support, additional implicit network permissions granted")
+	}
+
+	// Ensure that generated policy is what we computed above.
+	created, removed, err := osutil.EnsureDirState(dirs.SnapConfineAppArmorDir, glob, policy)
+	if err != nil {
+		return fmt.Errorf("cannot synchronize snap-confine policy, %s", err)
+	}
+	if len(created) == 0 && len(removed) == 0 {
+		// If the generated policy didn't change, we're all done.
+		return nil
+	}
+
+	// If snapd is executing from the core snap the it means it has
+	// re-executed. In that case we are no longer using the copy of
+	// snap-confine from the host distribution but our own copy. We don't have
+	// to re-compile and load the updated profile as that is performed by
+	// setupSnapConfineReexec below.
+	if strings.HasPrefix(exe, dirs.SnapMountDir) {
+		return nil
+	}
+
+	// Reload the apparmor profile of snap-confine. This points to the main
+	// file in /etc/apparmor.d/ as that file contains include statements that
+	// load any of the files placed in /var/lib/snapd/apparmor/snap-confine.d/.
+	// For historical reasons we may have a filename that ends with .real or
+	// not.  If we do then we prefer the file ending with the name .real as
+	// that is the more recent name we use.
+	var profilePath string
+	for _, profileFname := range []string{"usr.lib.snapd.snap-confine.real", "usr.lib.snapd.snap-confine"} {
+		profilePath = filepath.Join(dirs.SystemApparmorDir, profileFname)
+		if _, err := os.Stat(profilePath); err != nil {
+			if os.IsNotExist(err) {
+				continue
+			}
+			return err
+		}
+		break
+	}
+	if profilePath == "" {
+		return fmt.Errorf("cannot find system apparmor profile for snap-confine")
+	}
+
+	// We are not using apparmor.LoadProfile() because it uses other cache.
+	cmd := exec.Command("apparmor_parser", "--replace",
+		// Use no-expr-simplify since expr-simplify is actually slower on armhf (LP: #1383858)
+		"-O", "no-expr-simplify",
+		"--write-cache", "--cache-loc", dirs.SystemApparmorCacheDir,
+		profilePath)
+
+	if output, err := cmd.CombinedOutput(); err != nil {
+		// When we cannot reload the profile then let's remove the generated
+		// policy. Maybe we have caused the problem so it's better to let other
+		// things work.
+		osutil.EnsureDirState(dirs.SnapConfineAppArmorDir, glob, nil)
+		return fmt.Errorf("cannot reload snap-confine apparmor profile: %v", osutil.OutputErr(output, err))
+	}
+	return nil
+}
+
+var (
+	procSelfMountInfo = mount.ProcSelfMountInfo
+)
+
+// isHomeUsingNFS returns true if /home contains or might contain NFS mounts.
```

Comment:
The "might contain" is a truism and would be nice to clarify, in the sense that /etc/fstab _might_ contain NFS entries at all times. Clearly there's something more specific implied there since it mentions both "contains" and "might contain", though.

Reply:
Re worded to

`// isHomeUsingNFS returns true if NFS mounts are defined or mounted under /home.`